ddtrace/tracer/span.go

```diff
@@ -505,6 +505,7 @@ func shouldComputeStats(s *span) bool {
 // String returns a human readable representation of the span. Not for
 // production, just debugging.
 func (s *span) String() string {
+	s.RLock()
```

Comment:
👍 
While we're fixing this, can we also just defer the `RUnlock` for good measure?

Reply:
updated!
Let me know if there is more to do.

Reply:
Can you add a godoc comment on the mutex describing what is protected by the lock? It's unclear to me if it's just the Duration, or if it's all of the span types? Either way, let's just be explicit here.